euth/dashboard/forms.py

```diff
@@ -170,7 +281,215 @@ class Meta:
+                            answer['freetextAnswer'] = True
+                        else:
+                            answer['freetextAnswer'] = False
+
+                        answers.append(answer)
+                        a=a+1
+                        answer_key = "module_settings-question_"+str(q)+"_choice_"+str(a)+"_answerText"
+
+                    question['answers'] = answers
+                    questions.append(question)
+                    q=q+1
+                    question_key = "module_settings-question_"+str(q)+"_questionType"
+
+
+                jsonGenerator['questions'] = questions                                                   
+                poll = jsonGenerator                
+                self.data._mutable = True    
+                self.data['module_settings-poll'] = json.dumps(poll)
+            
+            
+            else:                    
+                # case edit            
+                if 'key' in self.initial:
+                    pollid = self.initial['key']
+                    if pollid:
+                    
+                        url_poll = '{base_url}/poll/{poll_id}'.format(
+                            base_url=settings.FLASHPOLL_BACK_URL,
+                            poll_id=pollid
+                        )
+
+                        # Handle get
+                        headers = {'Content-type': 'application/json'}
+                        response = requests.get(url_poll, headers=headers, auth=HTTPBasicAuth(settings.FLASHPOLL_BACK_USER, settings.FLASHPOLL_BACK_PASSWORD))        
+                        poll = json.loads(response.text)   
+                        poll = self.get_ordered_poll(poll)
+                        
+                        
+                else:
+                    # case create                                                    
+                    jsonstring = '{\"title\":\"\",\"shortDescription\":\"\",\"longDescription\":\"\",\"concludeMessage\":\"\",\"geofenceLocation\":\"\", \"startTime\":\"\", \"endTime\":\"\", \"descriptionMediaURLs\":[\"\"],\"keywords\":[\"\"],\"questions\":[{\"questionText\":\"\",\"orderId\":1,\"questionType\":\"CHECKBOX\",\"mandatory\":true,\"mediaURLs\":[\"\"],\"answers\":[{\"answerText\":\"\",\"orderId\":1,\"mediaURL\":\"\",\"freetextAnswer\":false},{\"answerText\":\"\",\"orderId\":2,\"mediaURL\":\"\",\"freetextAnswer\":false}]}]}'                    
+                    poll = json.loads(jsonstring)                    
+                        
+                        
+            self.fields['poll'] = forms.CharField(widget=forms.Textarea)
+            self.initial['poll'] = json.dumps(poll)
+                    
+            # description                    
+            self.fields['title'] = forms.CharField(label='Title')
+            self.initial['title'] = poll['title']
+            
+            self.fields['shortDescription'] = forms.CharField(widget=forms.Textarea, label='Subtitle')
```

Comment:
Why are you adding fields dynamically. If you would define a flashpoll form class, it would also reduce the amount of code that you need for parsing those form values.